File: formationJpa/src/main/java/formationJpa/entities/Adresse.java
package formationJpa.entities;

import javax.persistence.Column;
import javax.persistence.Embeddable;

@Embeddable
public class Adresse {
	@Column(name="num",length = 50)
	private String numero;
	private String rue;
	private String codePostal;
	private String ville;
	
	public Adresse() {
		
	}

	public Adresse(String numero, String rue, String codePostal, String ville) {
		this.numero = numero;
		this.rue = rue;
		this.codePostal = codePostal;
		this.ville = ville;
	}

	public String getNumero() {
		return numero;
	}

	public void setNumero(String numero) {
		this.numero = numero;
	}

	public String getRue() {
		return rue;
	}

	public void setRue(String rue) {
		this.rue = rue;
	}

	public String getCodePostal() {
		return codePostal;
	}

	public void setCodePostal(String codePostal) {
		this.codePostal = codePostal;
	}

	public String getVille() {
		return ville;
	}

	public void setVille(String ville) {
		this.ville = ville;
	}
	
	
}

File: angular/src/app/model/adresse.ts
export class Adresse {
  constructor(
    private _numero?: string,
    private _rue?: string,
    private _codePostal?: string,
    private _ville?: string
  ) {}

  public get numero(): string | undefined {
    return this._numero;
  }

  public set numero(value: string | undefined) {
    this._numero = value;
  }
  public get rue(): string | undefined {
    return this._rue;
  }

  public set rue(value: string | undefined) {
    this._rue = value;
  }
  public get codePostal(): string | undefined {
    return this._codePostal;
  }

  public set codePostal(value: string | undefined) {
    this._codePostal = value;
  }
  public get ville(): string | undefined {
    return this._ville;
  }

  public set ville(value: string | undefined) {
    this._ville = value;
  }

  public get adresse(): string {
    return (
      'methode:' +
      this.numero +
      ' ' +
      this.rue +
      ' ' +
      this.codePostal +
      ' ' +
      this.ville
    );
  }
}
